### PR TITLE
feat(crux-ui): add chip filters

### PR DESCRIPTION
**Pull request by bvektor**
- add chip filters to deployments, registries and nodes list pages

### CODE REVIEW

File: web/crux-ui/src/elements/dyo-chips.tsx [hunk 1]
```diff
@@ -1,27 +1,31 @@
 import assert from 'assert'
 import clsx from 'clsx'
+import useTranslation from 'next-translate/useTranslation'
 import { useState } from 'react'
 
 interface DyoChipsProps<T> {
   className?: string
   key?: React.Key
   choices: readonly T[]
+  isFilter?: boolean
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-chips.tsx:10]: Let's be explicit about the implication of this prop. Rename it to addAllOption or something

File: web/crux-ui/src/hooks/use-filters.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ export type EnumFilter<TEnum> = {
 export const enumFilterFor =
   <Item, Enum>(propertiesOf: PropertyValuesOf<Item>): FilterFunction<Item, EnumFilter<Enum>> =>
   (items: Item[], filter: EnumFilter<Enum>) => {
-    if (filter.enum === undefined) {
+    if (filter?.enum === undefined || filter?.enum === null) {
```
Comment (m8vago) [inline web/crux-ui/src/hooks/use-filters.ts:101]: is there a case where the filter is not declared?
if no then:
```
if (!filter.enum)
```

File: web/crux-ui/src/hooks/use-filters.ts [hunk 3]
```diff
@@ -51,14 +51,14 @@ export const useFilters = <Item, Filter>(options: UseFiltersOptions<Item, Filter
...
-  text: string
+  text?: string
 }
 
 export const textFilterFor =
   <Item>(propertiesOf: PropertyValuesOf<Item>): FilterFunction<Item, TextFilter> =>
   (items: Item[], filter: TextFilter) => {
     const text = filter?.text?.trim().toLowerCase()
-    if (text && text.length < 1) {
+    if (!text || (text && text.length < 1)) {
```
Comment (m8vago) [inline web/crux-ui/src/hooks/use-filters.ts:61]: dont need the `text &&` part

File: web/crux-ui/src/pages/products.tsx [hunk 4]
```diff
@@ -18,15 +18,15 @@ import useTranslation from 'next-translate/useTranslation'
 import { useRouter } from 'next/dist/client/router'
 import { useRef, useState } from 'react'
 
-export const PRODUCT_TYPE_FILTER_VALUES = ['simple', 'complex', 'all'] as const
+export const PRODUCT_TYPE_FILTER_VALUES = [null, 'simple', 'complex'] as const
```
Comment (polaroi8d) [inline web/crux-ui/src/pages/products.tsx:21]: Can we use `unknow` instead of `null`?

**Comment by robot9706** [comment]
On the version deployments screen the status filter is different from the deployments list:
image
Is this on purpose?

File: web/crux-ui/src/elements/dyo-filter-chips.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import useTranslation from 'next-translate/useTranslation'
+import DyoChips, { DyoChipsProps } from './dyo-chips'
+
+interface DyoFilterChipsProps<T> extends DyoChipsProps<T> {
+  addAllOption?: boolean
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-filter-chips.tsx:5]: is this ever false?

File: web/crux-ui/src/hooks/use-filters.ts (hunk 2 above)
Comment (m8vago) [reply web/crux-ui/src/hooks/use-filters.ts:101]: I would set all as a default value then, so we don't have to worry about the filter being null

Comment (m8vago) [reply web/crux-ui/src/hooks/use-filters.ts:101]: nevermind i got it

Comment (m8vago) [reply web/crux-ui/src/hooks/use-filters.ts:101]: however:

```
const { enum } = filter
if (!enum)
```

**Review by robot9706 — APPROVED** [review]
LGTM 😎

**Review by polaroi8d — APPROVED** [review]

**Pull request merged**